modules/core-components/src/main/java/org/mule/runtime/core/internal/processor/IdempotentRedeliveryPolicy.java

```diff
@@ -263,6 +265,11 @@ public CoreEvent process(CoreEvent event) throws MuleException {
         incrementCounter(messageId, createMessagingException(event, ex));
         throw ex;
       }
+    } catch (ObjectStoreException e) {
```

Comment:
what if there was another problem in the object store not related to the key being null?